src/map_extras.cpp

```diff
@@ -1399,17 +1454,121 @@ static void mx_fumarole( map &m, const tripoint &abs_sub )
 
 static void mx_portal_in( map &m, const tripoint &abs_sub )
 {
-    int x = rng( 5, SEEX * 2 - 6 ), y = rng( 5, SEEY * 2 - 6 );
-    m.add_field( {x, y, abs_sub.z}, fd_fatigue, 3, 0_turns );
-    fungal_effects fe( *g, m );
-    for( int i = x - 5; i <= x + 5; i++ ) {
-        for( int j = y - 5; j <= y + 5; j++ ) {
-            if( rng( 1, 9 ) >= trig_dist( x, y, i, j ) ) {
-                fe.marlossify( tripoint( i, j, abs_sub.z ) );
-                if( one_in( 15 ) ) {
-                    m.place_spawns( GROUP_NETHER_PORTAL, 1, i, j, i, j, 1, true );
+    const tripoint portal_location = { rng( 5, SEEX * 2 - 6 ), rng( 5, SEEX * 2 - 6 ), abs_sub.z };
+    const int x = portal_location.x;
+    const int y = portal_location.y;
+
+    switch( rng( 6, 6 ) ) {
```

Comment:
```suggestion
    switch( rng( 1, 6 ) ) {
```
Unless you only want to Mi-Go one, of course

Reply:
That's a debug random number generator 😄